### DISCUSSION TITLE
Forking

### DISCUSSION THREAD

**Discussion by icodus** [sheabunge/code-snippets#115] [General]
What would be a fair way to fork the great Code Snippets? I mean modifying CS strongly to fit a framework specific needs and change it's name to not messing with the current one. Where and what info on CS should I provide in the new plugin?
Congrats on great plugin and would be grateful for an answer .
Wiktor

**Comment by sheabunge** [comment]
The only real requirement is that you follow the conditions of the license. For Code Snippets v2, that is the [MIT license](https://github.com/sheabunge/code-snippets/blob/v2.14.1/license.txt), which requires you  include the text in that file in any forked versions of the plugin.

Code Snippets v3 is licensed under the [GPLv2 or later](https://github.com/sheabunge/code-snippets/blob/master/license.txt), the same license that WordPress core uses. The requirements here are similar – include the license and copyright in any forks – but also your fork needs to be licensed under the same license, and need to disclose the source code when redistributing it. 

If you're following those conditions, then it should all be fine. Of course, if you're able to contribute any applicable improvements that you make back to the main version of the plugin, that would be fantastic, but that's completely optional.

**Reply by icodus** [reply]
Thanks a lot for the straight answer and would be happy to contribute, of course.
